campaignion_tracking/package/src/ga4.js

```diff
@@ -451,11 +452,11 @@ export class GA4Tracker {
         currency: currencyCode,
         value: revenue,
         items: [{
-          item_name: product?.name,
-          item_id: product?.id,
-          price: product?.price,
-          item_variant: product?.variant,
-          quantity: product?.quantity,
+          item_name: product.name,
+          item_id: product.id,
+          price: product.price,
+          item_variant: product.variant,
+          quantity: product.quantity,
```

Comment:
This commit shows nicely that the generation of the product data from `eventData` and `this._context` could be done in one place and called from all 3 handlers. What do you think?

Reply:
It could. Also makes more sense to store the formatted data than the raw one. Have pushed a new commit.